lib/libutee/tee_api_operations.c

```diff
@@ -1265,12 +1265,12 @@ TEE_Result TEE_MACCompareFinal(TEE_OperationHandle operation,
 	if (res != TEE_SUCCESS)
 		goto out;
 
-	if (computed_mac_size != macLen) {
-		res = TEE_ERROR_MAC_INVALID;
+	if (macLen > computed_mac_size) {
```

Comment:
Should we consider the case for `macLen = 0`? If some implementations expect an error for it while others don't, then perhaps just leave it to the TA to handle?

Reply:
Ideally this case needs to be clarified by GP too, but as you said since it is not useful in practice it can be handled by the TA. The current code would return success I suppose, which totally makes sense IMO.